cumulus/client/cirrus-executor/src/lib.rs

```diff
@@ -622,7 +622,30 @@ where
 				let pre_state_root = as_h256(&execution_receipt.trace[local_trace_idx - 1])?;
 				let post_state_root = as_h256(&execution_receipt.trace[local_trace_idx])?;
 
-				let proof = todo!("Create `finalize_block` proof");
+				let block_builder = BlockBuilder::with_extrinsics(
+					&*self.client,
+					parent_header.hash(),
+					*parent_header.number(),
+					RecordProof::No,
+					Default::default(),
+					&*self.backend,
+					self.block_body(execution_receipt.secondary_hash)?,
+				)?;
+				let storage_changes =
+					block_builder.prepare_storage_changes_before_finalize_block()?;
+
+				let delta = storage_changes.transaction;
+				let post_delta_root = storage_changes.transaction_storage_root;
+
+				let proof = cirrus_fraud_proof::prove_execution(
+					&self.backend,
+					&*self.code_executor,
+					self.spawner.clone() as Box<dyn SpawnNamed>,
```

Comment:
I'm not sure `as Box<dyn SpawnNamed>` is actually needed here, definition seems to fit here.

Reply:
Otherwise the compiiler complains:

```
   Compiling subspace-node v0.1.0 (/home/xlc/src/github.com/subspace/subspace/crates/subspace-node)
   Compiling parachain-template-node v0.1.0 (/home/xlc/src/github.com/subspace/subspace/cumulus/parachain-template/node)
   Compiling cirrus-block-builder v0.1.0 (/home/xlc/src/github.com/subspace/subspace/cumulus/client/block-builder)
   Compiling cirrus-client-executor v0.1.0 (/home/xlc/src/github.com/subspace/subspace/cumulus/client/cirrus-executor)
error[E0277]: the trait bound `Box<dyn SpawnNamed + Sync + std::marker::Send>: SpawnNamed` is not satisfied
   --> cumulus/client/cirrus-executor/src/lib.rs:640:17
    |
640 |                 let proof = cirrus_fraud_proof::prove_execution(
    |                             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ the trait `SpawnNamed` is not implemented for `Box<dyn SpawnNamed + Sync + std::marker::Send>`
    |
    = help: the following implementations were found:
              <Box<(dyn SpawnNamed + 'static)> as SpawnNamed>
note: required by a bound in `cirrus_fraud_proof::prove_execution`
   --> /home/xlc/src/github.com/subspace/subspace/cumulus/client/fraud-proof/src/lib.rs:36:9
    |
36  |     Spawn: SpawnNamed + Send + 'static,
    |            ^^^^^^^^^^ required by this bound in `cirrus_fraud_proof::prove_execution`

For more information about this error, try `rustc --explain E0277`.
error: could not compile `cirrus-client-executor` due to previous error
warning: build failed, waiting for other jobs to finish...
error: build failed
```